alf/metrics/metrics.py

```diff
@@ -266,6 +279,68 @@ def _extract_metric_values(self, time_step):
             return [reward[..., i] for i in range(reward.shape[-1])]
 
 
+class AverageDiscountedEpisodicReturnMetric(AverageEpisodicSumMetric):
+    """Metric for computing the average discounted episodic return."""
+
+    def __init__(self,
+                 name='AverageDiscountedEpisodicReturnMetric',
+                 prefix='Metrics',
+                 reward_shape=(),
+                 dtype=torch.float32,
+                 discount=0.99,
+                 batch_size=1,
+                 buffer_size=10):
+        if reward_shape == ():
+            example_metric_value = torch.zeros((), device='cpu')
+        else:
+            example_metric_value = torch.zeros(
+                reward_shape, device='cpu').reshape(-1)
+            example_metric_value = list(example_metric_value)
+
+        self._discount = discount
+        self._accumulated_discount = torch.ones(batch_size, device='cpu')
+
+        super(AverageDiscountedEpisodicReturnMetric, self).__init__(
+            name=name,
+            dtype=dtype,
+            prefix=prefix,
+            batch_size=batch_size,
+            buffer_size=buffer_size,
+            example_metric_value=example_metric_value)
+
+    def _extract_metric_values(self, time_step):
```

Comment:
So this metric will only calculate the discounted return at the beginning of an episode. The values for RL algorithms usually means the average discounted return over all states. So that are not directly comparable.

Reply:
That is true. The current metric is episodic not directly comparable to the ``values`` curve.
I'm also trying to implementing a version by averaging over the states encountered during rollout.

Reply:
You'll probably need to record the value computed during loss computation, store it into some member variable of the algorithm, and then plot it (alf.summary.scalar/histogram) in summarize_rollout().

Reply:
> So this metric will only calculate the discounted return at the beginning of an episode. The values for RL algorithms usually means the average discounted return over all states. So that are not directly comparable.

Now changed the metric to be the average of all the states in rollout.

Reply:
> You'll probably need to record the value computed during loss computation, store it into some member variable of the algorithm, and then plot it (alf.summary.scalar/histogram) in summarize_rollout().

Calculating the value during rollout is probably the best to be comparable to the discounted return. This is not included currently as it will have some additional computational cost. 
Currently, we many get a coarse estimation from the value of the loss function.

Reply:
Ah, you are right, my suggestion won't work for rollout loss.